Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/htmlCreator.py

```diff
@@ -0,0 +1,189 @@
+
+    if time:
+        out += "<h1>High-level parameters versus time (IOV)</h1>\n"
+        # get list with names of the structures
+        for structure in [x.name for x in time if x.parameter == "xyz"]:
+            out += "<h2>{0}<h2>\n".format(structure)
+            for mode in ["xyz", "rot"]:
+                if any([x.filename for x in time if (x.parameter == mode and x.name == structure)]):
+                    filename = [x.filename for x in time if (x.parameter == mode and x.name == structure)][0]
+                    out += "<a href='plots/pdf/{0}.pdf'><img src='plots/png/{0}.png'></a>\n".format(
+                        filename)
+
+    # hole modules
+
+    # check if there are module plots
+    if any(x for x in config.outputList if (x.plottype == "mod" and x.number == "")):
+        out += "<h1>Module-level parameters</h1>\n"
+
+        # loop over all structures
+        for moduleName in [x.name for x in alignables.structures]:
+
+            # check if there is a plot for this module
+            if any(x for x in config.outputList if (x.plottype == "mod" and x.number == "" and x.name == moduleName)):
+                out += "<h2>{0}</h2>\n".format(moduleName)
+
+                # loop over modes
+                for mode in ["xyz", "rot", "dist"]:
+
+                    # get module plot
+                    module = [x for x in config.outputList if (
+                        x.plottype == "mod" and x.number == "" and x.name == moduleName and x.parameter == mode)]
+                    # get list of sub module plots
+                    moduleSub = [x for x in config.outputList if (
+                        x.plottype == "subMod" and x.number != "" and x.name == moduleName and x.parameter == mode)]
+
+                    # check if plot there is a plot in this mode
+                    if module:
+                        out += "<a href='plots/pdf/{0}.pdf'><img src='plots/png/{0}.png'></a>\n".format(module[
+                                                                                                        0].filename)
+
+                        # loop over submodules
+                        for plot in moduleSub:
+                            out += "<a href='plots/pdf/{0}.pdf'><img src='plots/png/{0}.png'></a>\n".format(
+                                plot.filename)
+
+    # plot taken from the millePedeMonitor_merge.root file
+
+    if any(x for x in config.outputList if x.plottype == "monitor"):
+        out += "<h1>Monitor</h1>\n"
+        for plot in [x for x in config.outputList if x.plottype == "monitor"]:
+            out += "<h3>{0}</h3>\n".format(plot.name)
+            out += "<a href='plots/pdf/{0}.pdf'><img src='plots/png/{0}.png'></a>\n".format(
+                plot.filename)
+
+    data = data.substitute(message=config.message, out=out)
+
+    with open(os.path.join(config.outputPath, outputFile), "w") as output:
+        output.write(data)
+        output.close()
```

Comment:
@prigoll you are using a 'with' construct which automatically closes the file, i.e. this line is redundant